File: food_delivery/src/main/java/com/datn/food_delivery/models/Shop.java
package com.datn.food_delivery.models;

public class Shop {
    private Long shop_id;
    private String name;
    private String address;
    private String imageURL;

    public Shop(Long shop_id, String name, String address, String imageURL) {
        this.shop_id = shop_id;
        this.name = name;
        this.address = address;
        this.imageURL = imageURL;
    }

    public Shop(Long shop_id, String name, String address) {
        this.shop_id = shop_id;
        this.name = name;
        this.address = address;
    }

    public Shop() {
    }

    public long getShop_id() {
        return shop_id;
    }

    public void setShop_id(long shop_id) {
        this.shop_id = shop_id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getImageURL() {
        return imageURL;
    }

    public void setImageURL(String imageURL) {
        this.imageURL = imageURL;
    }
}

File: food_delivery_flutter/lib/Model/Shop.dart
class Shop{
  final int shop_id;
  final String name;
  final String address;
  final String imageURL;

  Shop({required this.shop_id, required this.name, required this.address, required  this.imageURL});

  factory Shop.fromJson(Map<String, dynamic> json) {
    return Shop(
        shop_id: json['shop_id'],
        name: json['name'],
        address: json['address'],
        imageURL: json['imageURL'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'shop_id': shop_id,
      'name': name,
      'address': address,
      'imageURL': imageURL,
    };
  }

  factory Shop.toJson(Map<String, dynamic> json) {
    return Shop(
        shop_id: json['shop_id'],
        name: json['name'],
        address: json['address'],
        imageURL: json['imageURL'],
    );
  }
}